File: src/IPL_MANAGER/Player.java
package IPL_MANAGER;

/**
 * Player class representing a cricket player.
 */
class Player {
    private final String name;
    private int runsScored;
    private int wicketsTaken;
    private int fiftiesCount;

    public Player(String name, int runsScored, int wicketsTaken) {
        this.name = name;
        this.runsScored = runsScored;
        this.wicketsTaken = wicketsTaken;
        this.fiftiesCount = 0;
    }

    public String getName() {
        return name;
    }

    public int getRunsScored() {
        return runsScored;
    }

    public int getWicketsTaken() {
        return wicketsTaken;
    }

    public int getFiftiesCount() {
        return fiftiesCount;
    }

    public void addRunsScored(int runs) {
        runsScored += runs;
    }

    public void addWicketsTaken(int wickets) {
        wicketsTaken += wickets;
    }

    public void incrementFiftiesCount() {
        fiftiesCount++;
    }
}

File: src/ipl_manager_kotlin/Player.kt
package ipl_manager_kotlin

/**
 * Player class representing a cricket player.
 */
class Player(val name: String, private var runsScored: Int, private var wicketsTaken: Int) {
    private var fiftiesCount: Int = 0

    @JvmName("getPlayerName")
    fun getName(): String {
        return name
    }

    fun getRunsScored(): Int {
        return runsScored
    }

    fun getWicketsTaken(): Int {
        return wicketsTaken
    }

    fun getFiftiesCount(): Int {
        return fiftiesCount
    }

    fun addRunsScored(runs: Int) {
        runsScored += runs
    }

    fun addWicketsTaken(wickets: Int) {
        wicketsTaken += wickets
    }

    fun incrementFiftiesCount() {
        fiftiesCount++
    }
}
